core/tools.js

```diff
@@ -2069,6 +2071,43 @@
 				var ret = this.filter( array, fn, thisArg );
 
 				return array.length === ret.length;
+			},
+
+			/**
+			 * Tests whether any element in an array pass the test implemented by the provided function.
+			 * Returns `false` if the provided array is empty.
+			 *
+			 * ```javascript
+			 * var some = CKEDITOR.tools.array.some( [ 11, 2, 3, 4 ], function( value ) {
+			 * 	return value > 10;
+			 * } );
+			 * console.log( some );
+			 * // Logs: true
+			 * ```
+			 *
+			 * @param {Array} array
+			 * @param {Function} fn A function that gets called with each `array` item.
+			 * @param {Mixed} fn.value The currently iterated array value.
+			 * @param {Number} fn.index The index of the currently iterated value in an array.
+			 * @param {Array} fn.array The original array passed as the `array` variable.
+			 * @param {Mixed} [thisArg=undefined] A context object for `fn`.
+			 * @returns {Boolean} Information whether any element pass the test.
+			 * @member CKEDITOR.tools.array
+			 * @since 4.12.0
+			 */
+			some: function( array, fn, thisArg ) {
+				// Empty arrays always return false.
+				if ( array.length === 0 ) {
+					return false;
+				}
```

Comment:
It will still work correctly without this entire `if`, since `for` will not iterate and `false` will be eventually returned. WDYT about removing it?